src/utils/utils.h

```diff
@@ -225,8 +230,57 @@ struct Account {
 
   /// Move constructor.
   Account(uint256_t&& balance, uint64_t&& nonce) : balance(std::move(balance)), nonce(std::move(nonce)) {}
+
+  /// Deserialize constructor.
+  Account(const BytesArrView& bytes);
+
+  /// Serialize the account.
+  /// We serialize as balance + nonce + codeHash + contractType + code (if any)
+  /// 32 bytes + 8 bytes + 32 bytes + 1 byte + N (0 or more bytes) = 73 + N bytes
+  Bytes serialize() const;
+  bool isContract() const { return contractType != ContractType::NOT_A_CONTRACT; }
 };
 
+/**
+ * NonNullUniquePtr is a wrapper around std::unique_ptr that ensures the pointer is never null.
+ */
+template<typename T>
+class NonNullUniquePtr {
+private:
+  std::unique_ptr<T> ptr;
+
+public:
+  // Constructor that initializes the pointer with a new object if none is provided.
+  NonNullUniquePtr() : ptr(std::make_unique<T>()) {}
+
+  // Constructor that calls T<Ts...> with the provided arguments.
+  template<typename... Ts>
+  NonNullUniquePtr(Ts&&... args) : ptr(std::make_unique<T>(std::forward<Ts>(args)...)) {}
+
+  // Move constructor
+  NonNullUniquePtr(NonNullUniquePtr&& other) noexcept : ptr(std::move(other.ptr)) {}
```

Comment:
Your move constructor does exactly what a `default` would.
```suggestion
  NonNullUniquePtr(NonNullUniquePtr&&) = default;
```